### PR TITLE
Streamline date operations

### PR DESCRIPTION
**Pull request by justinmimbs**

### CODE REVIEW

**Comment by gregziegan** [comment]
Awesome!! Can you merge in my new changes from `master`? I pushed before seeing this PR 😄 

I'll start doing work on independent branches now too like a real git person 😜

**Comment by justinmimbs** [comment]
Absolutely! :)

**Comment by justinmimbs** [comment]
My forked master is now only 1 commit ahead of your master (instead of 1 ahead; 6 behind), so hopefully now the merge will just be a fast-forward. :)

File: src/Calendar.elm [hunk 1]
```diff
@@ -133,15 +134,16 @@ view state =
...
     let
-        firstOfWeek =
-            Date.Extra.floor Date.Extra.Week date
+        startOfWeek =
+            Date.Extra.floor Date.Extra.Sunday date
     in
         Date.Extra.range Date.Extra.Day
             1
-            (Date.Extra.floor Date.Extra.Sunday firstOfWeek)
-            (Date.Extra.ceiling Date.Extra.Sunday firstOfWeek)
```
Comment (justinmimbs) [inline src/Calendar.elm:144]: In Date.Extra, `floor` and `ceiling` work like they do on numbers, so if the given date is already at a rounded interval, then both functions will return the same date (e.g. `(floor 7.0, ceiling 7.0) == (7, 7)`). Instead, you want to `floor` for the start date, and then `add` to that for the end date.

File: src/Calendar.elm [hunk 2]
```diff
@@ -133,15 +134,16 @@ view state =
             ]
 
 
-dayRangeOfWeek date =
+weekRangeFromDate : Date -> List Date
+weekRangeFromDate date =
     let
-        firstOfWeek =
-            Date.Extra.floor Date.Extra.Week date
```
Comment (justinmimbs) [inline src/Calendar.elm:139]: Date.Extra uses Monday as the starting day of the week, so using `floor` or `ceiling` with `Date.Extra.Week` is synonymous to using it with Monday (which isn't what you want here).

File: src/Calendar.elm [hunk 3]
```diff
@@ -409,8 +367,8 @@ viewAgenda date =
         groupedEvents =
             eventsGroupedByDate events
 
-        isDateInMonth eventsDate =
-            Date.Extra.isBetween (Date.Extra.floor Date.Extra.Month date) (Date.Extra.ceiling Date.Extra.Month date) eventsDate
+        isDateInMonth =
+            Date.Extra.equalBy Date.Extra.Month date
```
Comment (justinmimbs) [inline src/Calendar.elm:371]: Hey, there's this for that! :)

**Comment by gregziegan** [comment]
sweet! Thanks!!

**Review by justinmimbs — COMMENTED** [review]
Hey Greg, thanks for using Date.Extra! I added some notes to this diff to point out some ways of using the library. If you have suggestions for the library, please let me know, as I'm happy to have user feedback.

**Pull request merged**